gpkit/constraints/set.py

```diff
@@ -127,19 +128,17 @@ def subinplace(self, subs, value=None):
         for constraint in self:
             constraint.subinplace(subs, value)
 
-    @property
-    def varkeys(self):
-        "return all Varkeys present in this ConstraintSet"
-        return self._varkeys()
-
-    def _varkeys(self, init_dict=None):
-        "return all Varkeys present in this ConstraintSet"
-        init_dict = {} if init_dict is None else init_dict
-        out = KeySet(init_dict)
+    def reset_varkeys(self, init_dict=None):
+        "Goes through constraints and collects their varkeys."
+        varkeys = KeySet()
+        if not init_dict is None:
+            varkeys.update(init_dict)
         for constraint in self:
             if hasattr(constraint, "varkeys"):
-                out.update(constraint.varkeys)
-        return out
+                varkeys.update(constraint.varkeys)
+            # else:
+            #     print type(constraint) # getting numpy bools...
+        self.varkeys = varkeys
```

Comment:
Next issue on this branch is with SummingConstraintSet, implemented as follows:

```python
class SummingConstraintSet(ConstraintSet):
    def __init__(self, lhs, varname, models=[], variables=[], **kwargs):
        summedvars = set([v.key for v in variables])
        alreadysummed = set()
        for model in models:
            mvars = model[varname]
            if not hasattr(mvars, "__len__"):
                mvars = [mvars]
            # next line makes the recursion stop at depth one
            # for safety to avoid double counting
            mvars = [v for v in mvars if v.key.models[0] == model.name]
            assert len(mvars) == 1
            summedvars = summedvars.union([v.key for v in mvars])
            for constraint in model.flat():
                if hasattr(constraint, "summedvars"):
                    alreadysummed = alreadysummed.union(constraint.summedvars)
        summedvars = summedvars.difference(alreadysummed)
        ConstraintSet.__init__(self, [lhs >= sum(Variable(**vk.descr)
                                                 for vk in summedvars)],
                               **kwargs)
```

When trying to initialize a SCS, get the following traceback:
```
     12             # next line makes the recursion stop at depth one
     13             # for safety to avoid double counting
---> 14             mvars = [v for v in mvars if v.key.models[0] == model.name]
     15             assert len(mvars) == 1
     16             summedvars = summedvars.union([v.key for v in mvars])

TypeError: 'NoneType' object has no attribute '__getitem__'
```